src/taipy/core/_repository/_repository_factory.py

```diff
@@ -8,22 +8,28 @@
 # Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
 # an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
 # specific language governing permissions and limitations under the License.
-
+from importlib import util
 from typing import Any, Optional, Union
 
+from src.taipy.core._repository import _FileSystemRepository, _SQLRepository
 from taipy.config import Config
 
-from .._repository import _FileSystemRepository
-
 
 class _RepositoryFactory:
```

Comment:
# Actual implementation
In the end, this class is not really a factory. It returns the base repository class (FileSystem or SQL) and not the final implementation. 
```
CycleRepo > FSRepo > AbstractRepo
CycleRepo > SQLRepo > AbstractRepo
CycleRepo > MongoRepo > AbstractRepo
```
The advantage is that we are ensuring by code that the 3 possible implementations of a CycleRepo have the exact same API. This is great.

However, my concern is that we cannot have a specific implementation for a query about cycles in the FS. The code must be either shared with all FSRepos, so nothing specific to cycle (if implemented in FSRepo) or common to all CycleRepos, so nothing specific to FS or SQL or Mongo (if implemented in CycleRepo). I have the feeling it will be difficult to benefit from the query system. 
Do you see it as a problem ? Maybe not.

# What I had in mind
I imagined that the factory could return the actual final implementation of a repo (like FsCycleRepo, SQLCycleRepo, or MongoCycleRepo).

The diagram would be something like that:
```
FsCycleRepo > CycleRepo > FSRepo > AbstractRepo
SQLCycleRepo > CycleRepo > SQLRepo > AbstractRepo
MongoCycleRepo > CycleRepo > MongoRepo > AbstractRepo
```

That would require a bit more code: 
- One factory method per implementation (cycle, data, job, ...)
- One child class per  (FsCycleRepo in the core repo and SQLCycleRepo and MongoCycleRepo in the enterprise repo).

The signature of the factory method would be like this:
`RepoFactory.getCycleRepo() -> CycleRepo`

Depending on the config, the factory would build a FsCycleRepo, an SQLCycleRepo, or a MongoCycleRepo. They all inherit from CycleRepo. And dynamically they would inherit either from FSRepo, either from SQLRepo or from MongoRepo.
I am not sure if/how it is feasible. And maybe it can be implemented later.
I am not even sure it is a better idea anyway.

What do you think?